writer.go

```diff
@@ -1439,7 +1456,19 @@ func (c *ColumnWriter) newColumnBuffer() ColumnBuffer {
 	case c.maxRepetitionLevel > 0:
 		column = newRepeatedColumnBuffer(column, c.maxRepetitionLevel, c.maxDefinitionLevel, nullsGoLast)
 	case c.maxDefinitionLevel > 0:
-		column = newOptionalColumnBuffer(column, c.maxDefinitionLevel, nullsGoLast)
+		// Since these buffers are pooled, we can afford to allocate a bit more memory in
+		// order to reduce the risk of needing to resize the buffer.
+		var (
+			size             = int(float64(column.Cap()) * 1.5)
+			rows             = int32Buffers.get(size)
+			definitionLevels = buffers.get(size)
+		)
+		c.closer = closerFunc(func() error {
+			rows.unref()
+			definitionLevels.unref()
+			return nil
+		})
+		column = newOptionalColumnBuffer(column, rows.data[:0], definitionLevels.data[:0], c.maxDefinitionLevel, nullsGoLast)
```

Comment:
Is there a reason why you don't apply the same strategy to allocating the repetition levels?